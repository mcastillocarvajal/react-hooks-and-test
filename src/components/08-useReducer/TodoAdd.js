import React from 'react'
import { useForm } from '../../hooks/useForm'

export const TodoAdd = ({ handleAddTodo }) => {

    const [ { description }, handleInputChange, reset] = useForm({
        description: '',
    })

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 0 ){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        }

        handleAddTodo( newTodo )
        reset();
    }

    return (
        <>
            <h4> Add todo </h4>
            <hr />

            <form onSubmit={ handleSubmit }>
                <input
                    className='form-control'
                    type='text'
                    name='description'
                    placeholder='Learn...'
                    autoComplete='off'
                    value={ description }
                    onChange={ handleInputChange }
                />

                <button 
                    className='btn btn-outline-primary mt-1 btn-block'
                    type='submit'>
                    Add
                </button>
            </form>
        </>
    )
}
