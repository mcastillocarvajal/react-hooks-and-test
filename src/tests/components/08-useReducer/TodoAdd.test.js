import React from 'react';
import { shallow } from 'enzyme';
import { TodoAdd } from '../../../components/08-useReducer/TodoAdd';
import { demoTodos } from '../../fixtures/demoTodos';

describe('test for <TodoAdd />', () => {

    const handleAddTodo = jest.fn();

    const wrapper = shallow(
        <TodoAdd
            handleAddTodo={ handleAddTodo }
        />
    )

    test('should show component correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    })

    test('should NOT call handleAddTodo ', () => {

        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault(){} });

        expect( handleAddTodo ).toHaveBeenCalledTimes(0);
    })

    test('should call function handleAddTodo', () => {
        
    })
    
    
})
