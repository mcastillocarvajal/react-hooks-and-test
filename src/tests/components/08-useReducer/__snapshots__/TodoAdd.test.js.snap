// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <TodoAdd /> should show component correctly 1`] = `
<Fragment>
  <h4>
     Add todo 
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
